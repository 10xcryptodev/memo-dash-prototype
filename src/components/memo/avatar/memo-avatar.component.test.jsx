import React from 'react'
import { shallow } from 'enzyme'
import { Comment } from 'semantic-ui-react'

import MemoAvatarComponent from './memo-avatar.component'
import testUsers from '../../../test-utils/test-users'

describe('<MemoAvatarComponent />', () => {
  let wrapper
  const alice = testUsers['alice']

  beforeEach(() => {
    const div = document.createElement('div')
    document.body.appendChild(div)
  })

  describe('should render', () => {
    it('without crashing', () => {
      wrapper = shallow(<MemoAvatarComponent memo={alice.memos[0]} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('own memo', () => {
      wrapper = shallow(<MemoAvatarComponent isMemoOfCurrentUser={true} memo={alice.memos[0]} />)
      expect(wrapper).toMatchSnapshot()
    })

    it('foreign memo', () => {
      wrapper = shallow(<MemoAvatarComponent isMemoOfCurrentUser={false} memo={alice.memos[0]} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('interaction', () => {
    it('should call onGoToProfileClicked when clicked on avatar', () => {
      const onGoToProfileClicked = jest.fn()
      wrapper = shallow(
        <MemoAvatarComponent
          onGoToProfileClicked={onGoToProfileClicked}
          isMemoOfCurrentUser={false}
          memo={alice.memos[0]}
        />
      )
      wrapper.find(Comment.Avatar).simulate('click')
      expect(onGoToProfileClicked).toHaveBeenCalled()
    })
  })
})
