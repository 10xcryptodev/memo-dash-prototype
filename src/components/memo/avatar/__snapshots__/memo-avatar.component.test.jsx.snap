// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MemoAvatarComponent /> should render foreign memo 1`] = `
<React.Fragment>
  <CommentAvatar
    as="a"
    onClick={[Function]}
  />
</React.Fragment>
`;

exports[`<MemoAvatarComponent /> should render own memo 1`] = `
<React.Fragment>
  <CommentAvatar
    className="avatar"
  />
</React.Fragment>
`;

exports[`<MemoAvatarComponent /> should render without crashing 1`] = `
<React.Fragment>
  <CommentAvatar
    as="a"
    onClick={[Function]}
  />
</React.Fragment>
`;
